### DISCUSSION TITLE
hotukdeals.com: nuisance

### DISCUSSION THREAD

**Discussion by ArchibaldToast** [uBlockOrigin/uAssets#26051] [Q&A] [answered]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367), [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) or [_a shortener/hosting site_](https://github.com/uBlockOrigin/uAssets/discussions/17361) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/user-attachments/assets/1879fc88-09bc-4de4-9e2f-3834a6ab396a), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin along with other content blockers.
- [X] I have verified that the web browser's built-in content blocker/tracking protection, [network wide/DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking), or my VPN is not causing the issue.
- [X] I have verified that other extensions are not causing the issue.
- [X] If this is about a breakage or detection, I have verified that it is caused by uBlock Origin and isn't a site issue.
- [ ] I did not answer truthfully to **ALL** the above checkboxes.

### URL address of the web page

`https://www.hotukdeals.com/`

### Category

nuisance

### Description

Hi,

Could anyone help with creating a custom filter for https://www.hotukdeals.com/ please. I'm looking to block deals from specific retailers, wherever they appear e.g. block all deals from Amazon and Nintendo eShop at https://www.hotukdeals.com/

I'd previously tried this - [www.hotukdeals.com##.thread-link.cept-tt:has-text(Amazon):has(.thread)](http://www.hotukdeals.com/##.thread-link.cept-tt:has-text(Amazon):has(.thread)) - but it no longer seems to work and didn't work at all for retailers with a space in their name, like 'Nintendo eShop'.

Thanks

### Other extensions used

none

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![01](https://github.com/user-attachments/assets/696d7c70-24ee-4a15-8699-d6ce35d8a749)

![02](https://github.com/user-attachments/assets/f664f9e3-96fe-4a7b-87ed-0400819b801e)
</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.61.0
Chromium: 130
filterset (summary):
 network: 138234
 cosmetic: 49731
 scriptlet: 23083
 html: 0
listset (total-discarded, last-updated):
 default:
  user-filters: 51-15, never
  easylist: 78846-45, now
  easyprivacy: 53271-671, now
  plowe-0: 3557-376, now
  ublock-badware: 11686-19, now
  ublock-filters: 40686-376, now
  ublock-privacy: 1448-24, now
  ublock-quick-fixes: 206-4, now
  ublock-unbreak: 2507-12, now
  urlhaus-1: 20475-0, 26m
filterset (user): [array of 51 redacted]
trustedset:
 added: [array of 74 redacted]
userSettings:
 advancedUserEnabled: true
 collapseBlocked: false
hiddenSettings: [none]
supportStats:
 allReadyAfter: 696 ms (selfie)
 maxAssetCacheWait: 229 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 1
 network:
  fuseplatform.net: 1
 extended:
  ##.ad-label
  ##.ad--center
  ###ad-mrec2
  ##.ad-sticky
```
</details>

**Comment by mapx-** [comment, marked as answer]
for amazon this filter is working on my end

```adb
hotukdeals.com##.thread:has-text(Amazon)
```

I see no nintendo

**Reply by ArchibaldToast** [reply]
Brilliant. Worked for both examples above. Thanks

**Reply by gorhill** [reply]
This one does not require procedural cosmetic filters, enforced natively by the browser:

    hotukdeals.com##.thread:has([data-t="merchantLink"][href$="=1650"])

---

Never mind, turns out not all "Amazon" links have an `href` property. I think this one is a bit more narrowed as it looks only for the link text:

    hotukdeals.com##.link[data-t="merchantLink"]:has-text(Amazon):upward(.thread)

If you want to target more retailers, you can use a regex instead:

    hotukdeals.com##.link[data-t="merchantLink"]:has-text(/Amazon|eBay/):upward(.thread)

**Reply by ArchibaldToast** [reply]
Thanks. The multiple retailers regex is especially useful.

**Reply by ArchibaldToast** [reply]
If I may also ask, what are the data-t names that relate to 'user/poster' and the 'item description'? I've trawled through the logger and can't see what these are identified as? Thanks

**Reply by gorhill** [reply]
Use the browser dev tools => Elements pane to find details about the DOM.

**Reply by garry-ut99** [reply]
> mapx-: https://github.com/uBlockOrigin/uAssets/discussions/26051#discussioncomment-11258535 : for amazon this filter is working on my end
> ```adb
> hotukdeals.com##.thread:has-text(Amazon)
> ```

it's too broad, it causes false hits because it matches "Amazon" text even in title or in description: 
[screenshoot A](https://github.com/user-attachments/assets/b9a1e106-1231-4eca-b50f-e2bd38c22e68) (in description)
[screenshoot B](https://github.com/user-attachments/assets/2a872751-3b2d-48b2-b513-dff29dfea624) (in description)
[screenshoot C](https://github.com/user-attachments/assets/5f3a328e-4817-4b74-8c90-21d8af0c582f) (in title) (<== I'm not sure whether it's desired to hide this one, but even if yes, there still might appear different cases which should not be hidden)

<hr>

> gorhill: https://github.com/uBlockOrigin/uAssets/discussions/26051#discussioncomment-11269845 : 
>
> This one does not require procedural cosmetic filters, enforced natively by the browser:
> 
> ```
> hotukdeals.com##.thread:has([data-t="merchantLink"][href$="=1650"])
> ```
> 
> Never mind, turns out not all "Amazon" links have an `href` property.

None of them have an `href` property until an user interacts with a link (`click`, `focus`, `pointerdown`), as there are `addEventListener` s attached, for example: middle-clicking a "link" to open in a new tab or clicking by right mouse button is sufficient to trigger `href` to appear (visible in DOM inspector), you must have interacted with some links, that's why `href` appeared in them.

<hr>

> gorhill: https://github.com/uBlockOrigin/uAssets/discussions/26051#discussioncomment-11269845 :
>
> I think this one is a bit more narrowed as it looks only for the link text:
> 
> ```
> hotukdeals.com##.link[data-t="merchantLink"]:has-text(Amazon):upward(.thread)
> ```
> 
> If you want to target more retailers, you can use a regex instead:
> 
> ```
> hotukdeals.com##.link[data-t="merchantLink"]:has-text(/Amazon|eBay/):upward(.thread)
> ```

Another alternative is:  

> ```adb
> hotukdeals.com##.thread:has(.link[data-t="merchantLink"]:has-text(Amazon))
> ```

just an alternative, I don't know which one is better, because I didn't benchmark them so far. I'm not saying any of them is better, or perhaps there is not a big difference, someone would have to determine it (unless you already did it). It's a good real world test case scenario for: https://github.com/gorhill/uBlock/wiki/Filter-Performance#upward-vs-has especially regarding _native_ `:has()` (as currenlty all major browsers support it), and not only _procedural_ `:has()`.
